### DISCUSSION TITLE
table_built event

### DISCUSSION THREAD

**Discussion by lewysigns** [anvilistas/tabulator#57] [Q&A]
Is there a way to tell if the table_built event has been raise?

I want to call a method, only if the table has been built. 

I could catch the runtime error in a try except, but I would rather check some ```self.tabulator.table_built``` flag.

Thanks!

**Comment by s-cork** [comment]
A bit more context please. 

Why would the table not be built?
Because of an error?

**Reply by lewysigns** [reply]
Sure thing. 

In my design, I have a data manager which manages communication between the client and the server, I then have subscribers to that data manager. This helps me use the same data across my application should multiple objects/tables/input/output forms need to read/write that data.

One of the subscribers has a tabulator table class object. When the data manager publishes to its subscribers, it attempts to update their table using the ```update_or_add_data()``` method. Sometimes the subscriber form has not been shown before this has occurred, which I believe is the cause of this error. (Since table_built event happens after the form show event --> I am reading this from the README.md and making inferences). I added more print statements for more context on the actions causing the error, but haven't been able to hit the error again.

**Reply by s-cork** [reply]
What about using a flag?
You can turn the flag off initially and then turn it on in the tableBuilt event. 

Alternatively I could add a custom error to catch. 

But there's no public flag available to check whether it's been built.

**Reply by s-cork** [reply]
There does look like tabulator has an `initialized` flag you could use. But it's not part of their docs so might be better to create your own flag.

**Reply by lewysigns** [reply]
Sounds good. 

Thank you! I'll go with creating my own flag.

Appreciate the guidance!

**Reply by mg3146** [reply]
Hi - I'm having a similar issue.  I'm trying to focus the table on the last row (using select row), but its throwing an error that tabulator isn't built yet, and to use "table_built"

**Reply by s-cork** [reply]
@mg3146 i think you've asked the same question on the forum? Assuming you've hooked up your table_built event to your event handler function I assume it's now working.

If not please provide more info. Ideally a clone link.

**Reply by mjmare** [reply]
I could use an initialised flag as well. This would help prevent code that depends on the tabulator state from running while initialisation is in progress. 

See: https://github.com/s-cork/Tabulator/issues/62

**Reply by lewysigns** [reply]
After posting and putting into practice, the table_built event is the way to.

You can use the add_event_handler to add the table_built event and put your selection logic in there.

Alternatively, you could put your init flag in the table built event.

**Reply by mjmare** [reply]
Why would this preclude an initialised flag? Of course I could do this myself, but that is repetitive. An initialised flag would be a slight burden for tabulator.
